src/BlockTypes/AddToCartForm.php

```diff
@@ -25,19 +25,27 @@ class AddToCartForm extends AbstractBlock {
 	 * @return string | void Rendered block output.
 	 */
 	protected function render( $attributes, $content, $block ) {
-		ob_start();
+		$post_id = $block->context['postId'] ?? 0;
+
+		if ( ! $post_id ) {
+			return;
+		}
+
+		$product = wc_get_product( $post_id );
 
-		while ( have_posts() ) {
```

Comment:
Do you mind sharing more details regarding this change? For the same reasons being discussed on https://github.com/woocommerce/woocommerce-blocks/discussions/8792#discussioncomment-5387577 , I would not recommend stop relying on `while ( have_posts() ) { the_post(); }`

Reply:
Hey @nefeline, I answered in the discussion post you shared, but just bringing an overview here. That was necessary because now the Single Product block can work in different contexts, for instance, in the Home template. Also, the merchants can select any product to be displayed and not only the product of the page they are accessing.

Reply:
Gotcha, thanks for clarifying! I think we can explore a different approach here: what do you think about [this proposal](https://github.com/woocommerce/woocommerce-blocks/pull/8610/files#r1148088334) ?